### ISSUE TITLE
Support of PackedSequence in SpikingLSTM

### ISSUE THREAD

**Issue by ShinoharaYuuyoru** [fangwei123456/SpikingFlow#130]
Hello!

I'm running `spiking_lstm_text.py` and I totally rewrote the code into `PyTorch machine learning style` :)
I'm testing the code now and I'll create a new PR if the code and the results are OK.

**My problem:**
As usual, we set `batch_size` to the dataset for training. However, I read the source code `~/clock_driven/rnn.py` and found that the code didn't support `PackedSequence` type so that we cannot train with batch unless crop or pad the different length sequences in one batch.
So, is there any plan to implement `PackedSequence` feature in spiking RNN/LSTM in the future?

Thank you all the time.

**Comment by fangwei123456** [comment]
> I read the code of two function you mentioned. Of course they solve the problem of padding the different length of sequences.
> It works well, however, I wonder using such a batch with padded sequences may cause the waste of the computing resources.

You are right, these codes are written by python. I will write a cupy backend for `padded_sequence_mask`. In most cases, `pad_sequence_collate` run on CPU, and I think it will not affect much about speed.

**Comment by fangwei123456** [comment]
```python
import torch
from spikingjelly import datasets
x1 = torch.rand([2, 6])
x2 = torch.rand([3, 6])
x3 = torch.rand([4, 6])
x = torch.nn.utils.rnn.pad_sequence([x1, x2, x3])  # [T, N, *]
print('x.shape=', x.shape)
x_len = torch.as_tensor([x1.shape[0], x2.shape[0], x3.shape[0]]).cuda()
mask = datasets.padded_sequence_mask(x_len)
print('mask.shape=', mask.shape)
print('mask=\\n', mask)
```

```
x.shape= torch.Size([4, 3, 6])
mask.shape= torch.Size([4, 3])
mask=\n tensor([[ True,  True,  True],
        [ True,  True,  True],
        [False,  True,  True],
        [False, False,  True]], device='cuda:0')
```
